pathdriver/path_driver.go

```diff
@@ -0,0 +1,79 @@
+package pathdriver
+
+import (
+	"path/filepath"
+)
+
+// PathDriver provides all of the path manipulation functions in a common
+// interface. The context should call these and never use the `filepath`
+// package or any other package to manipulate paths. The context should call
+// NormalizePath() before calling other other functions since some OSes accept
+// multiple path separators like Windows.
+type PathDriver interface {
+	Join(paths ...string) string
+	IsAbs(path string) bool
+	Rel(base, target string) (string, error)
+	Base(path string) string
+	Dir(path string) string
+	Clean(path string) string
+	Split(path string) (dir, file string)
+	Separator() byte
+	Abs(path string) (string, error)
+	Walk(string, filepath.WalkFunc) error
+	NormalizePath(path string) string
```

Comment:
Since this isn't a direct abstraction to filepath, make sure to document the expectations, or just call it `FromSlash`.